jetty-webapp/src/main/java/org/eclipse/jetty/webapp/StandardDescriptorProcessor.java

```diff
@@ -650,11 +651,10 @@ public void visitSessionConfig(WebAppContext context, Descriptor descriptor, Xml
         XmlParser.Node tNode = node.get("session-timeout");
         if (tNode != null)
         {
-            java.math.BigDecimal asDecimal = new java.math.BigDecimal(tNode.toString(false, true));
-            if (asDecimal.compareTo(org.eclipse.jetty.server.session.SessionHandler.MAX_INACTIVE_MINUTES) > 0)
-                throw new IllegalStateException("Max session-timeout in minutes is " + org.eclipse.jetty.server.session.SessionHandler.MAX_INACTIVE_MINUTES);
-
-            context.getSessionHandler().setMaxInactiveInterval(asDecimal.intValueExact() * 60);
+            int val = Integer.parseInt(tNode.toString(false, true));
+            if (val > SessionHandler.MAX_SESSION_TIMEOUT_MINS)
+                throw new IllegalStateException("Max session-timeout in minutes is " + SessionHandler.MAX_SESSION_TIMEOUT_MINS);
+            context.getSessionHandler().setMaxInactiveInterval(SessionHandler.clipSessionTimeout(val) * 60);
```

Comment:
I think this should just be `context.setSessionTimeout(val)` so that all the logic for checking the value is in `ServletContextHandler` and there is no need to expose constants and methods and duplicate logic.

Reply:
Solved this a different way, please see previous comment.